WooCommerce/Classes/ViewRelated/Themes/ThemesPreviewView.swift

```diff
@@ -0,0 +1,146 @@
+import SwiftUI
+
+struct ThemesPreviewView: View {
+    enum PreviewDevice: CaseIterable {
+        case mobile
+        case tablet
+        case desktop
+
+        static var defaultDevice: PreviewDevice {
+            return UIDevice.current.userInterfaceIdiom == .pad ? .tablet : .mobile
+        }
+
+        var width: CGFloat {
+            switch self {
+            case .mobile: return 400
+            case .tablet: return 800
+            case .desktop: return 1200
+            }
+        }
+
+        var viewportScript: String {
+            let js = """
+            // remove all existing viewport meta tags - some themes included multiple, which is invalid
+            document.querySelectorAll("meta[name=viewport]").forEach( e => e.remove() );
+            // create our new meta element
+            const viewportMeta = document.createElement("meta");
+            viewportMeta.name = "viewport";
+            viewportMeta.content = "width=%1$d";
+            // insert the correct viewport meta tag
+            document.getElementsByTagName("head")[0].append(viewportMeta);
+            """
+
+            return String(format: js, NSInteger(width))
+        }
+    }
+
+    @State private var selectedDevice: PreviewDevice? = PreviewDevice.defaultDevice
+
+    var body: some View {
+        NavigationView {
+            VStack(spacing: 0) {
+                WebView(
+                    isPresented: .constant(true),
+                    url: URL(string: "https://tsubakidemo.wpcomstaging.com/")!,
+                    previewDevice: $selectedDevice
+                )
+
+                Divider()
+                    .frame(height: Layout.dividerHeight)
+                    .foregroundColor(Color(.divider))
+
+                VStack {
+                    Button(Localization.startWithThemeButton) {
+                        /* todo */
+                    }
```

Comment:
Please add a callback closure to this view and trigger it when this button is tapped:

```suggestion
                    Button(Localization.startWithThemeButton, action: onStart)
```

Reply:
Also handled in https://github.com/woocommerce/woocommerce-ios/pull/11452/commits/f8cd5e8532a43dce31d484508b4a5c5cdc43f188